packages/vite/src/node/server/index.ts

```diff
@@ -902,19 +916,24 @@ export async function _createServer(
     }
   })
 
-  // serve static files under /public
-  // this applies before the transform middleware so that these files are served
-  // as-is without transforms.
-  if (publicDir) {
-    middlewares.use(servePublicMiddleware(server, publicFiles))
-  }
+  if (config.experimental.fullBundleMode) {
+    // serve memory output dist files
+    middlewares.use(memoryFilesMiddleware(server, false))
+  } else {
+    // serve static files under /public
+    // this applies before the transform middleware so that these files are served
+    // as-is without transforms.
+    if (publicDir) {
+      middlewares.use(servePublicMiddleware(server, publicFiles))
+    }
```

Comment:
`servePublicMiddleware` should be put before `memoryFilesMiddleware` and `transformMiddleware` and outside of this `if-else` so that files under `/public` works for full bundle mode.

Alternatively, we can put the public files in `server.memoryFiles`, but I guess it's more easier to handle if it's stored in a separate map.

Reply:
I think using the original `servePublicMiddleware` is better, using the `server.memoryFiles` is similar logic with `sirv`.